zend_parse_arg_impl.go

```diff
@@ -49,19 +53,8 @@ func zendParseArgAsString(value any) (string, error) {
 	var str string
 
 	switch v := value.(type) {
-	case string:
-		str = v
-	case int, int64:
-		str = fmt.Sprintf("%v", v)
-	case float64:
-		str = floatToString(v)
-	case bool:
-		// return "1" for true and an empty string("") for false
-		if v {
-			str = "1"
-		} else {
-			str = ""
-		}
+	case string, int, int64, float64, bool:
+		return ConvertToString(value)
```

Comment:
As I write below, the `zend_parse_arg_impl` function and `_convert_to_string` function work differently in different purpose. 

So, I think the Go cases corresponding the original cases like below should be handled differently.

```c 
case IS_OBJECT:
	if (parse_arg_object_to_string(arg, p, pl, IS_STRING TSRMLS_CC) == SUCCESS) {
		if (c == 'p' && CHECK_ZVAL_NULL_PATH(*arg)) {
			return "a valid path";
		}
		break;
	}

case IS_ARRAY:
case IS_RESOURCE:
default:
	return c == 's' ? "string" : "a valid path";
```

And I think... at here, you should also care how to handle some types which does not exist in php-src like `chan` and `map` things.

Reply:
The purpose of gophplib is to port functions frm PHP to Go, ensuring that identical inputs yield identical outputs. Therefore, to support only types available in PHP, excluding those unique to Go. However, as maps in Go are equivalent to associative arrays in PHP that guarantees order, I have utilized the [orderedmap third-party library ](https://pkg.go.dev/github.com/elliotchance/orderedmap/v2)to support ordered maps in implode function.

Reply:
Ok thanks, and... just in case, I think you should support a 'map' base type for unordered cases, because if order of map is not important, then user can use like `Implode(map)`.

Reply:
modified to support regular maps as well. Additionally, I've added a comment indicating that the order cannot be guaranteed for map types